src/main/java/com/slack/kaldb/chunk/ReadWriteChunkImpl.java

```diff
@@ -148,6 +151,10 @@ public boolean snapshotToS3(String bucket, String prefix, S3BlobFs s3BlobFs) {
     } catch (Exception e) {
       LOG.error("Exception when copying RW chunk " + chunkInfo + " to S3.", e);
       return false;
+    } finally {
+      if (indexCommit != null) {
```

Comment:
It's better to move the null check into the function instead of the call site.

Reply:
makes sense!